pennylane/ops/qubit/hamiltonian.py

```diff
@@ -342,7 +342,7 @@ def compute_grouping(self, grouping_type="qwc", method="rlf"):
                 self.ops, grouping_type=grouping_type, method=method
             )
 
-    def simplify(self):
+    def simplify(self):  # pylint: disable=arguments-differ
```

Comment:
Maybe instead we could do something like:

```suggestion
    def simplify(self, depth=None):
    
    if depth is not None:
        return self
```

That way we can still call this method in the conventional way, but we can also handle something like:

```
sum(qml.PauliX(0), H).simplify()
```

Reply:
Do we want to call this `simplify` method when simplifying nested operators? Soran mentioned that this method is super ineffective.

**Note:** Calling `sum(qml.PauliX(0), H).simplify()` won't crash becaus the `simplify` method of the sum calls `H.simplify(depth=depth)`, which corresponds to the `simplify` method defined in the `Operator` class.

Reply:
> Soran mentioned that this method is super inefficient.

There is a version of [simplify](https://github.com/PennyLaneAI/pennylane/blob/c868132deef9c474a3422f1ca485251a29db1be0/pennylane/qchem/observable_hf.py#L200) inside qchem that is more efficient.

Reply:
```
(1.0 * qml.PauliX(0)).simplify(depth=2)
```
Does crash.

We want to make it so that if a `Hamiltonian` is called without a depth keyword argument, like `H.simplify()`, it does the expected simplification in-place. But, if it's called *with* a depth keyword argument, then we just `return self`.

Reply:
We should have a unified logic and only one `simplify` function though.

Reply:
Depth has been removed from `simplify` method.